### ISSUE TITLE
Removable Pill component doesn't meet accessibility requirements

### ISSUE THREAD

**Issue by mannycarrera4** [Workday/canvas-kit#1614]
## 🐛 Bug Report

When a pill is `removable` there is no connection between the text and the button. We need to restructure some aria attributes to make this accessible.
